ddtrace/contrib/botocore/services/bedrock.py

```diff
@@ -167,28 +167,33 @@ def _extract_text_and_response_reason(ctx: core.ExecutionContext, body: Dict[str
     provider = ctx["model_provider"]
     try:
         if provider == _AI21:
-            text = body.get("completions")[0].get("data").get("text")
-            finish_reason = body.get("completions")[0].get("finishReason")
+            completions = body.get("completions", [])
+            if completions:
+                text = body.get("completions")[0].get("data").get("text")
+                finish_reason = body.get("completions")[0].get("finishReason")
         elif provider == _AMAZON and "embed" in model_name:
             text = [body.get("embedding", [])]
         elif provider == _AMAZON:
-            text = body.get("results")[0].get("outputText")
-            finish_reason = body.get("results")[0].get("completionReason")
+            results = body.get("results", [])
+            if results:
+                text = results[0].get("outputText")
+                finish_reason = results[0].get("completionReason")
         elif provider == _ANTHROPIC:
             text = body.get("completion", "") or body.get("content", "")
             finish_reason = body.get("stop_reason")
         elif provider == _COHERE and "embed" in model_name:
             text = body.get("embeddings", [[]])
         elif provider == _COHERE:
-            text = [generation["text"] for generation in body.get("generations")]
-            finish_reason = [generation["finish_reason"] for generation in body.get("generations")]
+            generations = body.get("generations", [])
+            text = [generation["text"] for generation in generations]
```

Comment:
agree. just putting this PR as a quick fix for now to prevent crashes in general, but we definitely should invest some time to response parsing and validation for all of our current and future llm/ai integrations